### PR TITLE
BAEL-4399:Changes for showing NoSuchFieldError

### PR DESCRIPTION
**Pull request by amitiw4u**
Changes to show NoSuchError

### CODE REVIEW

File: spring-security-modules/spring-security-web-react/src/main/webapp/WEB-INF/view/react/package.json [hunk 1]
```diff
@@ -16,5 +16,17 @@
   },
   "devDependencies": {
     "eslint-plugin-react": "^7.9.1"
+  },
```
Comment (KevinGilmore) [inline spring-security-modules/spring-security-web-react/src/main/webapp/WEB-INF/view/react/package.json:19]: This file should not be part of this PR

File: core-java-modules/core-java-exceptions-3/src/main/java/com/baeldung/exceptions/nosuchfielderror/NoSuchFieldError.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.baeldung.exceptions.nosuchfielderror;
+
+import java.lang.reflect.Field;
+
+public class NoSuchFieldError {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-exceptions-3/src/main/java/com/baeldung/exceptions/nosuchfielderror/NoSuchFieldError.java:5]: This might get confused with the core Java class of the same name. See Javadoc https://docs.oracle.com/javase/8/docs/api/java/lang/NoSuchFieldError.html

File: core-java-modules/core-java-exceptions-3/src/main/java/com/baeldung/exceptions/nosuchfielderror/NoSuchFieldError.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+		Class<?> aClass = Class.forName(clss.getClass().getName());
+		Field field = aClass.getDeclaredField("message");
+		field.setAccessible(true);
+		String msgStr = (String) field.get(clss);
+		return msgStr;
+	}
+
+	public static String getDependentMessage() {
+		return Dependent.message;
+	}
```
Comment (KevinGilmore) [inline core-java-modules/core-java-exceptions-3/src/main/java/com/baeldung/exceptions/nosuchfielderror/NoSuchFieldError.java:20]: Indent with spaces instead of tabs

File: core-java-modules/core-java-exceptions-3/src/test/java/com/baeldung/exceptions/nosuchfielderror/NoSuchFieldErrorTest.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	@Test(expected = NoSuchFieldException.class)
+	public void whenDependentFieldNotFound_thenThrowNoSuchFieldException() {
+		NoSuchFieldError.getDependentMessage();
+	}
+	
+	@Test
+	public void whenDependentFieldNotFound_returnMessage() {
+		String dependentMessage = NoSuchFieldError.getDependentMessage();
+		assertTrue("Hello Baeldung".equals(dependentMessage));
+	}
```
Comment (KevinGilmore) [inline core-java-modules/core-java-exceptions-3/src/test/java/com/baeldung/exceptions/nosuchfielderror/NoSuchFieldErrorTest.java:23]: Indent with spaces, not tabs

File: core-java-modules/core-java-exceptions-3/src/test/java/com/baeldung/exceptions/nosuchfielderror/NoSuchFieldErrorTest.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package com.baeldung.exceptions.nosuchfielderror;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+public class NoSuchFieldErrorTest {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-exceptions-3/src/test/java/com/baeldung/exceptions/nosuchfielderror/NoSuchFieldErrorTest.java:7]: Unit test class names should end in "UnitTest" as per guidelines. Otherwise, the PR will fail the build.

**Review by KevinGilmore — CHANGES_REQUESTED** [review]
Rename unit test class. Rename NoSuchFieldError class. Indent with spaces. Remove file that shouldn't be part of this PR.

File: core-java-modules/core-java-exceptions-3/src/main/java/com/baeldung/exceptions/nosuchfielderror/NoSuchFieldError.java (hunk 2 above)
Comment (amitiw4u) [reply core-java-modules/core-java-exceptions-3/src/main/java/com/baeldung/exceptions/nosuchfielderror/NoSuchFieldError.java:5]: agreed. Will rename it.
